src/observable/ConnectableObservable.ts

```diff
@@ -25,7 +25,16 @@ export class ConnectableObservable<T> extends Observable<T> {
     return (this.subject = this.subjectFactory());
   }
 
-  connect(): Subscription<T> {
+  connect(onSubscribe?: (subscription: Subscription<T>) => void): Subscription<T> {
+    if (onSubscribe) {
+      this._callbackConnect(onSubscribe);
+      return null;
```

Comment:
In the callback version? I don't know. What should that be? Where would it be used?